lib/datadog/core/configuration/settings.rb

```diff
@@ -233,13 +233,6 @@ def initialize(*_)
             # Disable gathering of names and versions of gems in use by the service, used to power grouping and
             # categorization of stack traces.
             option :code_provenance_enabled, default: true
-
-            # Use legacy transport code instead of HttpTransport. Temporarily added for migration to HttpTransport,
-            # and will be removed soon. Do not use unless instructed to by support.
-            option :legacy_transport_enabled do |o|
-              o.default { env_to_bool('DD_PROFILING_LEGACY_TRANSPORT_ENABLED', false) }
-              o.lazy
-            end
```

Comment:
Note/alternative here: To avoid backwards-incompatible changes to the config (even though this should be a really rare configuration that customers should not use unless instructed by support), we could leave it in place until 2.x, and leave here a deprecation warning stated it doesn't do anything anymore.

Reply:
I've decided to adopt the alternative suggested -- in the latest version of the PR, the setting is left in place with a warning, rather than removed.